code/controllers/subsystem/monitor.dm

```diff
@@ -0,0 +1,131 @@
+SUBSYSTEM_DEF(monitor)
+	name = "Monitor"
+	init_order = INIT_ORDER_MONITOR
+	runlevels = RUNLEVEL_GAME
+	wait = FREQUENCY_STATE_CALCULATION
+	///The next world.time for wich the monitor subsystem refresh the state
+	var/scheduled = 0
+	///The current state
+	var/current_state = STATE_BALANCED
+	///The last state
+	var/last_state = STATE_BALANCED
+	///If we consider the state as a stalemate
+	var/stalemate = FALSE
+	///The current state points. Negative means xenos are winning, positive points correspond to marine winning
+	var/current_points = 0
+	///The number of time we had the same state consecutively
+	var/stale_counter = 0
+	///The number of humans on ground
+	var/human_on_ground = 0
+	///The number of humans being in either lz1 or lz2
+	var/human_in_FOB = 0
+	///The number of time most of humans are in FOB consecutively
+	var/humans_all_in_FOB_counter = 0
+	///TRUE if we detect a state of FOB hugging
+	var/FOB_hugging = FALSE
+	///TRUE if xenos have hijacked into ship. Disable the monitor
+	var/hijacked = FALSE
+	///List of all int stats
+	var/list/list/monitor_statistics[B17_USED]
+	
+
+
+/datum/controller/subsystem/monitor/Initialize(time, zlevel)
+	var/i
+	var/list/l
+	for(i=ANCIENT_QUEEN, i<=OB_AVAILABLE, i++)
+		monitor_statistics[i] = 0
+	for(i=MINIGUN_USED, i<= B17_USED, i++)
+		l = list()
+		monitor_statistics[i] = l
+	if(CONFIG_GET(flag/monitor_disallowed))
+		can_fire = 0
+	scheduled = START_STATE_CALCULATION + world.time
+	return ..()
+
+/datum/controller/subsystem/monitor/fire(resumed = 0)
+	if(!resumed & world.time>START_STATE_CALCULATION)
+		check_state() //only check these if we aren't resuming a paused fire
```

Comment:
shouldnt  START_STATE_CALCULATION be a VARSET_CALLBACK timer that sets can_fire to TRUE after the round starts?

Reply:
Done